controllers/managedocs_controller.go

```diff
@@ -977,16 +1062,45 @@ func (r *ManagedOCSReconciler) reconcilePrometheus() error {
 			Key: alertRelabelConfigSecretKey,
 		}
 
-		// prometheus tls -- will be uncommented
-		// tlsConfig := r.prometheus.Spec.Web.TLSConfig
-		// tlsConfig.KeySecret.LocalObjectReference.Name = prometheusServiceSecretName
-		// tlsConfig.Cert.Secret.LocalObjectReference.Name = prometheusServiceSecretName
-		// tlsConfig.ClientCA.ConfigMap.LocalObjectReference.Name = servingCertCABundleConfigMapName
+		if deployerCSV, err := r.getCSVByPrefix(deployerCSVPrefix); err == nil {
+			deployerCSVDeployment := &deployerCSV.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[0]
+			deployerCSVContainers := deployerCSVDeployment.Spec.Template.Spec.Containers
+			var kubeRbacImage string
+			for key := range deployerCSVContainers {
+				container := deployerCSVContainers[key]
+				if container.Name == "kube-rbac-proxy" {
+					kubeRbacImage = container.Image
+				}
+			}
+
+			prometheusContainers := r.prometheus.Spec.Containers
+			for key := range prometheusContainers {
+				container := &prometheusContainers[key]
+				if container.Name == "kube-rbac-proxy" {
+					container.Image = kubeRbacImage
+				}
+			}
```

Comment:
If the container already exists why are we forcing a specific kube rbac proxy? Why can't we use the rbac proxy that is already there?

Reply:
This was a mistake, I intended to add containerImage to the prometheus template and not to the runtime CR.
`desired := templates.PrometheusTemplate.DeepCopy()
prometheusContainers := desired.Spec.Containers`